host.go

```diff
@@ -289,6 +289,28 @@ func (h *gpbftRunner) receiveCertificate(c *certs.FinalityCertificate) error {
 	log.Infow("skipping forwards based on cert", "from", currentInstance, "to", nextInstance)
 
 	nextInstanceStart := h.computeNextInstanceStart(c)
+
+	// Look for any existing messages in WAL for the next instance, and if there is
```

Comment:
Hm. I guess technically we'll see this on start. Although we'll never see it for instance 0.

Any reason not to replay everything immediately in `gpbftRunner.Start`?

Reply:
Or we can look at the current instance at that point.

Reply:
That would take more gymnastics to get the current instance 🤷 

What I have done in the latest changes is to refactor calls to participant's `StartInstanceAt`, and replay message for the instance being started every time. 

Please let know what you think.

Reply:
Oh, that's much nicer.